notification/app/notification/services/azure/NotificationsHubSender.scala

```diff
@@ -34,23 +36,24 @@ abstract class NotificationsHubSender(
       platformStatistics = recipientsCount map { PlatformStatistics(WindowsMobile, _) }
     )
 
-    if (shouldSendToApps(push.notification)) {
-      for {
-        result <- send(push)
-        count <- count(push.destination)
-      } yield {
-        result.fold(
-          e => NotificationRejected(NotificationHubSenderError(e.some).some).left,
-          id => report(id, count.toOption).right
-        )
-      }
-    } else {
-      Future.successful(report(None, None).right)
+    converter.toRawPush(push) match {
```

Comment:
This looks like it's only needed if shouldSendToApps(push.notification) == true. 
I guess if it's not too much extra processing it's worth it to simplify this code